cli/tests/test_next_prev_commands.rs

```diff
@@ -567,6 +567,126 @@ fn test_next_editing() {
+    test_env.jj_cmd_ok(&repo_path, &["edit", "description(first)"]);
+    std::fs::write(&file_path, "first").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["new", "description(third)"]);
+    test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "fourth"]);
+    let (stdout, stderr) = test_env.jj_cmd_ok(&repo_path, &["prev", "--conflict"]);
+    // We now should be a child of `third`.
+    insta::assert_snapshot!(stdout, @"");
+    insta::assert_snapshot!(stderr, @r###"
+    Working copy now at: vruxwmqv 5050288d (conflict) (empty) (no description set)
+    Parent commit      : rlvkpnrz 7c39847a (conflict) second
+    There are unresolved conflicts at these paths:
+    content.txt    2-sided conflict
+    "###);
+}
+
+#[test]
+fn test_prev_conflict_editing() {
+    // Edit the third commit.
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+    let file_path = repo_path.join("content.txt");
+    test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "first"]);
+    test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "second"]);
+    std::fs::write(&file_path, "second").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "third"]);
+    // Create a conflict in the third commit, where we'll jump to.
+    test_env.jj_cmd_ok(&repo_path, &["edit", "description(first)"]);
+    std::fs::write(&file_path, "first text").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["new", "description(third)"]);
+    test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "fourth"]);
+    let (stdout, stderr) = test_env.jj_cmd_ok(&repo_path, &["prev", "--conflict", "--edit"]);
+    // We now should be editing the third commit.
+    insta::assert_snapshot!(stdout, @"");
+    insta::assert_snapshot!(stderr, @r###"
+    Working copy now at: kkmpptxz 26b1439f (conflict) third
+    Parent commit      : rlvkpnrz 55b5d11a (empty) second
+    There are unresolved conflicts at these paths:
+    content.txt    2-sided conflict
+    "###);
+}
+
+#[test]
+fn test_next_conflict() {
+    // There is a conflict in the third commit, so after next it should be the new
+    // parent.
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+    let file_path = repo_path.join("content.txt");
+    test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "first"]);
+    std::fs::write(&file_path, "second").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "second"]);
+    // Create a conflict in the third commit.
+    test_env.jj_cmd_ok(&repo_path, &["edit", "description(first)"]);
+    std::fs::write(&file_path, "first").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["new", "description(second)"]);
+    test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "third"]);
+    test_env.jj_cmd_ok(&repo_path, &["new", "description(first)"]);
```

Comment:
Could you create this commit on top of the second commit instead? I think the test case would then fail because you dropped the `.minus()` from my suggestion in https://github.com/martinvonz/jj/pull/3248/commits/b21081f92f90e84a22365aaee7a78ec002b6041b#r1597713740.

Reply:
pushed a version with it, it does not fail.

Reply:
Oops, sorry, I had missed a `.children()` in the revset just before `.descendants()`.

Reply:
This was not fixed. I'll send a PR to fix it.